### PR TITLE
Completed login functionality and style

### LINKED ISSUE #108: Login to Split
**User Story**
For example: 
As a user
I want to be able to log into my account
So I can use Split to track and save my bills

**Priority**
[x]: High
[]: Medium
[]: Low

**Describe the solution you want to implement**
Front-end needs to be hooked up to the back end and make the appropriate API calls to log in the user.

**Describe alternatives you've considered**
n/a

**Issue Dependencies**
#23, #44
also related to sign ups #96, but not completely dependant

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by babe269**
Closes #108 

**Description**

* Added the login functionality and matched style with the sign up page
* Added API integration with back-end for login.
* Created error message for bad login.

**Testing**

Checked login functionality is working correctly by testing valid and invalid credentials.

**Checklist**

- [x] I have tested my change
- [x] Code builds successfully and all tests pass
- [x] Docs have been added/updated if needed

### CODE REVIEW

File: split-webapp/split/src/App.css [hunk 1]
```diff
@@ -188,6 +188,10 @@
   margin-left: 5%;
 }
 
+.userTextField {
```
Comment (williamlixu) [inline split-webapp/split/src/App.css]: nit: PascalCase

File: split-webapp/split/src/App.css [hunk 2]
```diff
@@ -280,7 +284,7 @@
 
 .MuiTextField-root {
   margin: 5px !important;
-  width: 200px;
+  width: 200;
```
Comment (williamlixu) [inline split-webapp/split/src/App.css]: Leaving the px here is a better practice so it is clear what the units are. Think we accidentality removed it on the rebase.

File: split-webapp/split/src/pages/Login.js [hunk 3]
```diff
@@ -9,106 +8,169 @@ import "../App.css";
...
-      margin: theme.spacing(1),
-      width: 200,
-      background: "white"
-    }
-  },
-  margin: {
-    margin: theme.spacing(1)
+class Login extends Component {
+  constructor(props) {
+    // assign props and give null/ appropriate values to required fields
```
Comment (williamlixu) [inline split-webapp/split/src/pages/Login.js]: nit: don't really need this comment

File: split-webapp/split/src/pages/Login.js [hunk 4]
```diff
@@ -9,106 +8,169 @@ import "../App.css";
...
-                  src={mainLogo}
-                  style={{ width: "50%", marginTop: "10%" }}
-                  alt="Main logo for login"
-                />
-                <Box component="div" className="InnerContainer">
-                  <Typography component="h3" className="SignIn">
-                    Sign In
-                  </Typography>
-                  <form className={classes.root} noValidate autoComplete="off">
+  // hadles live changes in the username field
```
Comment (williamlixu) [inline split-webapp/split/src/pages/Login.js]: Pretty obvious code which means you chose good function names! Don't really need the comments (good thing!)

**Review by williamlixu — CHANGES_REQUESTED** [review]
Just few small things, looks great!

File: split-webapp/split/src/pages/Login.js [hunk 5]
```diff
@@ -9,106 +8,169 @@ import "../App.css";
...
+  handleOnChangeUser = event => {
+    console.log("Click");
+    this.state.username = event.target.value;
+  };
+
+  // handles live changes in the password field
+  handleOnChangePassword = event => {
+    console.log("Click");
```
Comment (jake-good) [inline split-webapp/split/src/pages/Login.js]: Might want to remove this line as unneeded now

**Review by jake-good — APPROVED** [review]
Overall looks good, some unneeded console logs that should eventually be removed but may still be useful for debugging

**Comment by babe269** [comment]
Fixed comment, css and logging issues.

**Review by williamlixu — APPROVED** [review]
Thanks for those changes, looks great!

**Pull request merged**